tests/test_tokenization.py

```diff
@@ -286,6 +286,294 @@ def test_bert_base_multilingual_bn(self):
+        self.assertEqual(['rabbi', '##gu', 'wu', '##xu', '##u', 'amar', 'ku', 'bi', '##xi', '##ye', '##y', 'in', 'la', 'dum', '##iy', '##o', 'qal', '##cada', '##ha', 'kan', '##ca', '##an', '.'], tokens)
+
+        tokens = tokenizer.tokenize('bixiyey')
+        self.assertEqual(['bi', '##xi', '##ye', '##y'], tokens)
+
+        tokenizer = BertTokenizer.from_pretrained('bert-base-multilingual-cased')
+
+        tokens = tokenizer.tokenize('Rabbigu wuxuu amar ku bixiyey in la dumiyo qalcadaha Kancaan.')
+        self.assertEqual(['Rabbi', '##gu', 'w', '##ux', '##uu', 'amar', 'ku', 'bi', '##xi', '##ye', '##y', 'in', 'la', 'dum', '##iyo', 'q', '##al', '##cada', '##ha', 'Kan', '##ca', '##an', '.'], tokens)
+
+        tokens = tokenizer.tokenize('bixiyey')
+        self.assertEqual(['bi', '##xi', '##ye', '##y'], tokens)
+    
+    def test_xlmr_base_multilingual_so(self):
+        """
+        Somali
+        """
+        tokenizer = AutoTokenizer.from_pretrained('xlm-roberta-base')
+
+        tokens = tokenizer.tokenize('Rabbigu wuxuu amar ku bixiyey in la dumiyo qalcadaha Kancaan.')
+        self.assertEqual(['▁Rabbi', 'gu', '▁wuxuu', '▁amar', '▁ku', '▁bixi', 'yey', '▁in', '▁la', '▁dum', 'iyo', '▁qal', 'cada', 'ha', '▁Kan', 'ca', 'an', '.'], tokens)
+
+        tokens = tokenizer.tokenize('bixiyey')
+        self.assertEqual(['▁bixi', 'yey'], tokens)
+
+    def test_bert_base_multilingual_sw(self):
+        """
+        Swahili
+        """
+        tokenizer = BertTokenizer.from_pretrained('bert-base-multilingual-uncased')
+
+        tokens = tokenizer.tokenize('Huduma ya upasuaji mkubwa na mdogo')
+        self.assertEqual(['hu', '##dum', '##a', 'ya', 'up', '##asu', '##aji', 'mk', '##ubwa', 'na', 'md', '##ogo'], tokens)
+
+        tokens = tokenizer.tokenize('upasuaji')
+        self.assertEqual(['up', '##asu', '##aji'], tokens)
+
+        tokenizer = BertTokenizer.from_pretrained('bert-base-multilingual-cased')
+
+        tokens = tokenizer.tokenize('Huduma ya upasuaji mkubwa na mdogo')
+        self.assertEqual(['Hu', '##dum', '##a', 'ya', 'up', '##asu', '##aji', 'mk', '##ub', '##wa', 'na', 'm', '##dogo'], tokens)
+
+        tokens = tokenizer.tokenize('upasuaji')
+        self.assertEqual(['up', '##asu', '##aji'], tokens)
+    
+    def test_xlmr_base_multilingual_sw(self):
+        """
+        Swahili
+        """
+        tokenizer = AutoTokenizer.from_pretrained('xlm-roberta-base')
+
+        tokens = tokenizer.tokenize('Huduma ya upasuaji mkubwa na mdogo')
+        self.assertEqual(['▁Huduma', '▁ya', '▁up', 'asu', 'aji', '▁mkubwa', '▁na', '▁mdogo'], tokens)
+
+        tokens = tokenizer.tokenize('upasuaji')
+        self.assertEqual(['▁up', 'asu', 'aji'], tokens)
+
+    def test_bert_base_multilingual_ti(self):
+        """
+        Tigrinya
+        """
+        tokenizer = BertTokenizer.from_pretrained('bert-base-multilingual-uncased')
+
+        tokens = tokenizer.tokenize('ስርዓተ ቀብሪ ኢንጂነር ስመኘው በቀለ ትማሊ ተፈፂሙ')
```

Comment:
@keleog all `[UNK]`? So what do we do here? Like, how is any downstream NLP task possible then...?

Reply:
We could adapt the tokenizer to the language via various methods (eg. https://www.aclweb.org/anthology/2020.findings-emnlp.240/).

Otherwise, we cannot use it on any downstream task. E.g. Recall our Japanese NER problem, and mBERT scores 0 on Amharic NER https://arxiv.org/abs/2103.11811

Reply:
Yup, good to know at least!